### ISSUE TITLE
VMware VM not closed at the end

### ISSUE THREAD

**Issue by Tigzy** [cuckoosandbox/cuckoo#2632]
I found the probable issue, in vmware.py: 
There's a else without a if: 

![image](https://user-images.githubusercontent.com/9531438/52127916-df1e4000-2633-11e9-803a-f6c6090f40a9.png)

**Comment by reox** [comment]
No this is not a lonly else :D 

> The try … except statement has an optional else clause, which, when present, must follow all except
> clauses. It is useful for code that must be executed if the try clause does not raise an exception. 

See https://docs.python.org/3/tutorial/errors.html

**Comment by Tigzy** [comment]
Ah I didn't know thanks :)
This is weird logic cause the Exception would leave the function execution anyway. 
Putting this else here has no other effect, we could just execute it's content after the try/except block. 

Anyway, I think the "issue" was rather a lack of documentation here.

**Comment by reox** [comment]
short excourse in python here ;) no, this makes sense, take a look at this code:
```
try:
   foo = some_function_that_might_fail()
except:
   print("okay this did not work")
else:
   some_other_function(foo)
```

This is actually not the same as 
```
try:
   foo = some_function_that_might_fail()
except:
   print("okay this did not work")
some_other_function(foo)
```
because in this case `foo` might have not been initialized.

You could obviously use 
```
try:
   foo = some_function_that_might_fail()
   some_other_function(foo)
except:
   print("okay this did not work")
```
But this could lead to problems if `some_other_function` raises the same exception as `some_function_that_might_fail` and you need to distinguish.

So the `else` block is used to execute code if and only if the full block in the try has finished.
btw there is also `finally`: https://docs.python.org/3/tutorial/errors.html#defining-clean-up-actions

**Comment by Tigzy** [comment]
I agree will all of this, the thing is that in your example there's no "raise Exception". That makes all the difference because in the Except block the function will continue to execute, unlike our Cuckoo function
